### DISCUSSION TITLE
How to draw border circle, not filled ?

### DISCUSSION THREAD

**Discussion by jdemaria22** [hajimehoshi/ebiten#2063] [Q&A] [answered]
HI guys , 

I need to create a circle not filled (only borders), 

I was investigating and i have this code that create the circle:

func DrawCircle(screen *ebiten.Image, x float32, y float32, radius float32, clr color.RGBA) {
	var path vector.Path

	path.MoveTo(x, y)
	path.Arc(x, y, radius, 0, math.Pi*2, vector.Clockwise)

	vertices, indices := path.AppendVerticesAndIndicesForFilling(nil, nil)

	redScaled := float32(clr.R) / 255
	greenScaled := float32(clr.G) / 255
	blueScaled := float32(clr.B) / 255
	alphaScaled := float32(clr.A) / 255

	for i := range vertices {
		v := &vertices[i]

		v.ColorR = redScaled
		v.ColorG = greenScaled
		v.ColorB = blueScaled
		v.ColorA = alphaScaled
	}

	screen.DrawTrianglesShader(vertices, indices, simpleShader, &ebiten.DrawTrianglesShaderOptions{
		FillRule: ebiten.EvenOdd,
	})

	screen.DrawTrianglesShader(vertices, indices, simpleShader, nil)
}

	simpleShader, err = ebiten.NewShader([]byte(`
		package gui

		func Fragment(position vec4, texCoord vec2, color vec4) vec4 {
			return color
		}
	`))

What do I have wrong in the code that i'ts  draw fill included?

Thanks!

**Comment by hajimehoshi** [comment, marked as answer]
What about using two circles by calling `Arc` twice so that the path will be a donut?

**Comment by jdemaria22** [comment]
Thanks for answering,

I need a circle like this:

![image](https://user-images.githubusercontent.com/43354137/162583852-52327400-8914-4c9e-bc44-38c18870cdf0.png)


If i call  Arc twice, draw a circle inside another?, but i need the inside of the circle to be transparent

**Reply by hajimehoshi** [reply]
Yes, this is what I intended and what the even-odd mode does.

**Reply by jdemaria22** [reply]
Thanks !!
That worked!!
